### PR TITLE
Fixing docstrings and imports in `quadrature`

### PR DESCRIPTION
**Pull request by mmahsereci**
*Issue #, if available:*

*Description of changes:*
This PR looks scary, but let me explain (all changes only involve the `quadrature` package):
- I finally fixed all docstrings in `quadrature` as they were incomplete, partially wrong, did not render correctly, or did not adhere to the right formatting. Things look pretty and informative now.
- I moved a few imports one level up. It's way more intuitive now. (e.g. instead of `from emukit.quadrature.loop.specific_loop_name import SpecificLoop` we now have `from emukit.quadrature.loop import SpecificLoop` etc). The latter imports were possible before already in most places, so it's more of a "hiding away" of the lengthy versions of it and making it consistent. Actual imports in code will only break occasionally in case the lengthy version was used. I think it's way more user-friendly like this. There might be other ways of doing this though.
- Some objects got their own submodule as it made sense. These are `emukit.quadrature.kernels.measures` for integration measures and `emukit.quadrature.loop.point_calculators` for quadrature specific point calculator.
- Renamed `BoxBounds` to `BoxDomain`. This object is only used internally, so should not interfere with much.
- Functionality of the code is unchanged. So this PR is just cosmetics, albeit a bit overdue...

@apaleyes perhaps you can check how the docs look. I hope the changes in module/import structure are OK for the quad package considering that it's easier to parse now for users. But let me know what you think.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: emukit/test_functions/quadrature/hennig1D.py [hunk 1]
```diff
@@ -10,22 +10,21 @@
 
 
 def hennig1D() -> Tuple[UserFunctionWrapper, List[Tuple[float, float]]]:
-    """
-    1D toy integrand coined by Philipp Hennig
+    r"""1D toy integrand coined by Philipp Hennig.
```
Comment (apaleyes) [inline emukit/test_functions/quadrature/hennig1D.py:13]: Is there a reference?

**Review by apaleyes — APPROVED** [review]
Fantastic work, Maren!

File: emukit/quadrature/acquisitions/squared_correlation.py [hunk 2]
```diff
@@ -11,55 +11,61 @@
...
-    SquaredCorrelation is identical to the integral-variance-reduction acquisition up to a global normalizing constant!
+    r"""The squared correlation between the integral value and integrand evaluations
+    under a Gaussian process model.
 
     .. math::
-        \rho^2(x) = \frac{(\int k_N(x_1, x)\mathrm{d}x_1)^2}{\mathfrac{v}_N v_N(x)}\in [0, 1]
+        a(x) = \frac{(\int k_N(s, x)p(s)\mathrm{d}s)^2}{\mathfrak{v}_N v_N(x)}\in [0, 1]
+
+    where :math:`\mathfrak{v}_N` is the current integral variance given :math:`N` observations,
+    :math:`v_N(x)` is the predictive integral variance if point :math:`x` was added newly,
```
[the remark this answers is not in the document]
Comment (mmahsereci) [reply emukit/quadrature/acquisitions/squared_correlation.py:21]: "was added" should be correct here, as we may not add the point. a(x) describes the acquisition value that is equal to the predictive integral variance if x was added to the model as evaluation location (in the sense of "were to be added hypothetically".)

File: emukit/test_functions/quadrature/hennig1D.py (hunk 1 above)
Comment (mmahsereci) [reply emukit/test_functions/quadrature/hennig1D.py:13]: Yes, there are many. I added one.

File: emukit/quadrature/acquisitions/squared_correlation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmahsereci) [reply emukit/quadrature/acquisitions/squared_correlation.py:21]: I am not a native speaker though...

**Pull request merged**